src/prefect/runner/submit.py

```diff
@@ -35,10 +36,30 @@ async def get_current_run_count() -> int:
         return response.json()
 
 
+async def run_prefect_callable_and_retrieve_run_id(
+    prefect_callable: Union[Flow, Task],
+    parameters: Dict[str, Any],
+) -> uuid.UUID:
+    flow_run_id: Optional[uuid.UUID] = None
```

Comment:
```suggestion
    """
    This function exists only to capture the flow_run_id that is created 
    while directly executing a flow. It does so by injecting its own 
    on_completion callback. This should only be called if flow run 
    submission to the Runner's webserver failed.
    """
    flow_run_id: Optional[uuid.UUID] = None
```

Cookie crumb trail to clarify why we're calling this sometimes. Maybe overly verbose as is?